olive/model/model_config.py

```diff
@@ -12,10 +12,10 @@
 class IOConfig(ConfigBase):
     # TODO remove input names, shapes and types, turn to use olive dataset config.
     input_names: List[str]
-    input_shapes: List[List[int]] = None
+    input_shapes: List[List[Union[int, str]]] = None
```

Comment:
Why is `str` added? For the IOConfig use case, it will always need to be `int`.

Reply:
To suppot the case where io_config generated from OnnxModel.

Reply:
I see. Thanks!

Reply:
In previous commit, I unified the `get_io_config` to return `IOConfig` which met failure for onnx model. But it is fine now, I change the `IOConfig` to `dict`.